components/concept/push/src/main/java/mozilla/components/concept/push/PushProcessor.kt

```diff
@@ -6,74 +6,95 @@
 
 package mozilla.components.concept.push
 
-import androidx.annotation.CallSuper
+import androidx.annotation.VisibleForTesting
 
 /**
  * A push notification processor that handles registration and new messages from the [PushService] provided.
  * Starting Push in the Application's onCreate is recommended.
  */
-abstract class PushProcessor {
+interface PushProcessor {
 
     /**
-     * Initialize the PushProcessor before it
+     * Start the push processor and any service associated.
      */
-    @CallSuper
-    open fun initialize() {
-        instance = this
-    }
-
-    /**
-     * Start the push processor and starts any service associated.
-     */
-    abstract fun start()
+    fun initialize()
 
     /**
-     * Stop the push service and stops any service associated.
+     * Removes all push subscriptions from the device.
      */
-    abstract fun stop()
+    fun shutdown()
 
     /**
      * A new registration token has been received.
      */
-    abstract fun onNewToken(newToken: String)
+    fun onNewToken(newToken: String)
 
     /**
      * A new push message has been received.
      */
-    abstract fun onMessageReceived(message: PushMessage)
+    fun onMessageReceived(message: EncryptedPushMessage)
 
     /**
      * An error has occurred.
      */
-    abstract fun onError(error: Error)
+    fun onError(error: PushError)
 
     companion object {
+        /**
+         * Initialize the PushProcessor that needs to be called before
+         */
+        fun install(processor: PushProcessor) {
```

Comment:
I'm not really sure there's value in having this singletone stuff live at the concept level. I'd expect to see it in `feature-push`.

Reply:
Every time I ask someone, I keep moving it to the location they prefer until the next review. :)

I think this is fine at the concept level since it's a necessity for a PushProcessor to be installed into the lifecycle of the Application regardless of the feature that implements it.

Reply:
Sounds good. You own this, it's your call ;)